### PR TITLE
Remove account and address discovery when wallet is created

### LINKED ISSUE #320: Account and address discovery should not occur when a wallet is created with a new seed (still required for restored seeds)
This will improve the UX when creating a brand new wallet because there will be less waiting.  Addresses will be able to be immediately generated since wallet won't need to search the entire blockchain for previous address usage.

### PR DESCRIPTION
**Pull request by vctt94**
fixes #320

### CODE REVIEW

**Comment by vctt94** [comment]
rebased

File: chain/sync.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func NewRPCSyncer(w *wallet.Wallet, rpcClient *RPCClient) *RPCSyncer {
 // needed to fully register the wallet for notifications and synchronize it with
 // the dcrd server are performed.  Otherwise, it will listen for notifications
 // but not register for any updates.
-func (s *RPCSyncer) Run(ctx context.Context, startupSync bool) error {
+func (s *RPCSyncer) Run(ctx context.Context, startupSync bool, isNewWallet bool) error {
```
Comment (jrick) [inline chain/sync.go]: can't do this, this is a breaking api change.

File: dcrwallet.go [hunk 2]
```diff
@@ -462,7 +462,21 @@ func rpcClientConnectLoop(ctx context.Context, passphrase []byte, jsonRPCServer
...
 		// reconnect.
 		syncer := chain.NewRPCSyncer(w, chainClient)
-		err = syncer.Run(ctx, true)
+		isNewWallet, err := loader.IsNewWallet()
+		if err != nil {
+			syncLog.Errorf("Wallet synchronization stopped: %v", err)
+			return
+		}
+		if isNewWallet {
+			defer func() {
```
Comment (jrick) [inline dcrwallet.go]: this runs in a loop, so this won't do as you want here.

File: loader/loader.go [hunk 3]
```diff
@@ -194,8 +195,9 @@ func (l *Loader) CreateWatchingOnlyWallet(extendedPubKey string, pubPass []byte)
 
 // CreateNewWallet creates a new wallet using the provided public and private
 // passphrases.  The seed is optional.  If non-nil, addresses are derived from
-// this seed.  If nil, a secure random seed is generated.
-func (l *Loader) CreateNewWallet(pubPassphrase, privPassphrase, seed []byte) (w *wallet.Wallet, err error) {
+// this seed.  If nil, a secure random seed is generated. If the wallet is just
+// created, it creates a new file at the dbdirpath.
+func (l *Loader) CreateNewWallet(pubPassphrase, privPassphrase, seed []byte, imported bool) (w *wallet.Wallet, err error) {
```
Comment (jrick) [inline loader/loader.go]: another unacceptable api change

**Comment by vctt94** [comment]
rebased

**Comment by vctt94** [comment]
Test is failing because of chain module: `github.com/decred/dcrwallet/chain`. Should I add a replace to the wallet to fix that? 

`replace github.com/decred/dcrwallet/wallet => ../wallet`

Also, just in order to compare. First syncing a wallet now on master: 

```
2019-02-08 16:28:21.661 [INF] WLLT: The wallet has been unlocked without a time limit
...
2019-02-08 16:30:00.436 [INF] SYNC: Blockchain sync completed, wallet ready for general usage.
```
Total time: 2 min 21 seg

At this branch: 

```
2019-02-08 16:22:35.338 [INF] WLLT: The wallet has been unlocked without a time limit
...
2019-02-08 16:23:40.119 [INF] SYNC: Blockchain sync completed, wallet ready for general usage.
```

Total time: 1 min 5 seg 


Full log master: https://pastebin.com/GQt43aNG
Full log this branch: https://pastebin.com/uvvD4xMs

**Comment by jrick** [comment]
I realize it was my comment in the issue that led to the additional file, but I think we need to take a different approach where this is tracked by the database instead.  I would also like it to track the birthday so we know where to start rescans and discovery from, which means we can't remove the additional file to signal that we've "caught up".

**Pull request closed without merging**